spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarProperties.java

```diff
@@ -936,6 +943,11 @@ public void setTimeout(Duration timeout) {
 			this.timeout = timeout;
 		}
 
+		void validate(String prefix) {
+			Assert.state((this.enabled || !this.required), "If transactions are required "
```

Comment:
`InvalidConfigurationPropertyValueException` should be used to report this sort of problem.